### PR TITLE
Improve input validation for `ensureClosingSlash` utility function.

### PR DESCRIPTION
**Pull request by SandipGyawali**
udpated the ensureClosingShalsh function with some check condition that throws error based on the input type validation and empty string.

signed-off-by: sandipgyawali10@gmai.com

### CODE REVIEW

**Comment by SandipGyawali** [comment]
@SerhiiYahdzhyiev #68

File: src/utils.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { TabsMapUpdater } from "./types";
 
 export function ensureClosingSlash(url: string): string {
+  if (!url.trim()) throw new Error("url input cannot be empty");
+
```
Comment (SerhiiYahdzhyiev) [inline src/utils.ts:5]: I would prefer the error message to be: "Invalid input: " + url.

Also please extend the condition check with: `typeof url !== "string"`.

**Comment by SerhiiYahdzhyiev** [comment]
Also please format your commit message. For reference check [this](https://gist.github.com/finalfantasia/bd0070673ca27e5f7473).

Also don't forget to run `lint` and `format` scripts before pushing. Maybe in future I will add corresponding git hooks to the repo directly.

**Comment by SandipGyawali** [comment]
@SerhiiYahdzhyiev okay

**Comment by SandipGyawali** [comment]
> Also please format your commit message. For reference check [this](https://gist.github.com/finalfantasia/bd0070673ca27e5f7473).
> 
> Also don't forget to run `lint` and `format` scripts before pushing. Maybe in future I will add corresponding git hooks to the repo directly.

updated as you mentioned on the folllowing lines.

**Review by SerhiiYahdzhyiev — CHANGES_REQUESTED** [review]

File: src/utils.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { TabsMapUpdater } from "./types";
 
 export function ensureClosingSlash(url: string): string {
+  if (typeof url == "string" && !url.trim())
```
[the remark this answers is not in the document]
Comment (SerhiiYahdzhyiev) [reply src/utils.ts:4]: If you are serious about the contributions, take time to read the corresponding sections in README and CONTRIBUTING docs.

**Comment by SerhiiYahdzhyiev** [comment]
Closing as not yet relevant.

**Pull request closed without merging**